sw/host/opentitanlib/src/transport/ultradebug/uart.rs

```diff
@@ -89,4 +89,14 @@ impl Uart for UltradebugUart {
         }
         Ok(())
     }
+
+    fn clear_rx_buffer(&self) -> Result<()> {
+        // Note: Want to call `purge_usb_rx_buffer()` but our `libftdi` is too old.
+        // safe_ftdi::Device::purge_usb_rx_buffer: error: undefined reference to 'ftdi_tciflush'
+        // Keep reading while until the RX buffer is empty.
+        const TIMEOUT: Duration = Duration::from_millis(5);
+        let mut buf = [0u8; 256];
+        while self.read_timeout(&mut buf, TIMEOUT)? > 0 {}
+        Ok(())
+    }
```

Comment:
We have two of the same implementation.  Should we create a default implementation in the trait (since it requires nothing apart from another trait method `read_timeout`) and then override in backends where we can do better?

Reply:
Done, thanks for catching this!